### PR TITLE
feat: Add timeout-setting to Graylog-plugin

### PR DESCRIPTION
**Pull request by SebastianThorn**
### Required for all PRs:

- [X] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [ ] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves [#10185](https://github.com/influxdata/telegraf/issues/10185)

I added timeout, but not sure if I did it correctly in regards to a default-vaule. I have not dealt with go before.

### CODE REVIEW

**Comment by SebastianThorn** [comment]
!signed-cla

**Comment by SebastianThorn** [comment]
!signed-cla

File: plugins/inputs/graylog/graylog.go [hunk 1]
```diff
@@ -30,10 +31,11 @@ type Metric struct {
...
 type GrayLog struct {
-	Servers  []string
-	Metrics  []string
-	Username string
-	Password string
+	Servers         []string
+	Metrics         []string
+	Username        string
+	Password        string
+	ResponseTimeout config.Duration
```
Comment (srebhan) [inline plugins/inputs/graylog/graylog.go:38]: When you are touching this anyway, can you please add the corresponding `toml` tag annotations?

**Review by srebhan — COMMENTED** [review]
@SebastianThorn  thanks for the submission. The code looks good. I just have two very minor comments about the option name and ask for your help to add toml-tags to the config-option fields...

File: plugins/inputs/graylog/README.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ Note: if namespace end point specified metrics array will be ignored for that ca
     "http://[graylog-server-ip]:12900/system/metrics/multiple",
   ]
 
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
```
[the remark this answers is not in the document]
Comment (SebastianThorn) [reply plugins/inputs/graylog/README.md:34]: Absolutly, there were different names in different plugins when i copied the code.

File: plugins/inputs/graylog/graylog.go (hunk 1 above)
Comment (SebastianThorn) [reply plugins/inputs/graylog/graylog.go:38]: Yes

**Comment by SebastianThorn** [comment]
@srebhan Does this look better?

File: plugins/inputs/graylog/graylog.go [hunk 3]
```diff
@@ -34,6 +35,7 @@ type GrayLog struct {
 	Metrics  []string
 	Username string
 	Password string
+	Timeout  config.Duration
```
Comment (srebhan) [inline plugins/inputs/graylog/graylog.go:38]: Can you please add the `toml` tags to the struct fields that receive user-options like
```suggestion
	Timeout  config.Duration `toml:"timeout"`
```
This makes it more clear which fields are configured by the user...

**Comment by srebhan** [comment]
The option looks good, I was waiting for the `toml` tags. Do you mind to add those?

**Review by srebhan — APPROVED** [review]
Ok looks like we should add the TOML tags later and take what we have.
Looks good to me. Thanks @SebastianThorn for your contribution!

**Review by powersj — APPROVED** [review]

**Comment by powersj** [comment]
> Ok looks like we should add the TOML tags later and take what we have. Looks good to me. Thanks @SebastianThorn for your contribution!

Can one of you please file a bug so we do not forget to go back and do this please?

Thanks!

**Comment by srebhan** [comment]
@powersj done in #10320.

**Pull request merged**